docs/data-sources/key_ring.md

```diff
@@ -0,0 +1,76 @@
+---
+# Documentation generated by "gen/generator.go"; DO NOT EDIT.
+# In order to regenerate this file execute `go generate` from the repository root.
+# More details can be found in the [README](https://github.com/CiscoDevNet/terraform-provider-aci/blob/master/README.md).
+subcategory: "Generic"
+layout: "aci"
+page_title: "ACI: aci_key_ring"
+sidebar_current: "docs-aci-data-source-aci_key_ring"
+description: |-
+  Data source for Key Ring
+---
+
+# aci_key_ring #
+
+Data source for Key Ring
+
+## API Information ##
+
+* Class: [pkiKeyRing](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/pkiKeyRing/overview)
+
+* Supported in ACI versions: 1.0(1e) and later.
+
+* Distinguished Name Formats:
+  - `uni/tn-{name}/certstore/keyring-{name}`
+  - `uni/userext/pkiext/keyring-{name}`
+
+## GUI Information ##
+
+* Location: `Generic`
+
+## Example Usage ##
+
+```hcl
+
+data "aci_key_ring" "example" {
+  name = "test_name"
+}
+
+data "aci_key_ring" "example_tenant" {
+  parent_dn = aci_tenant.example.id
+  name      = "test_name"
+}
+
+```
+
+## Schema ##
+
+### Required ###
+
+* `name` (name) - (string) The name of the Key Ring object.
+
+### Optional ###
+
+* `parent_dn` - (string) The distinguished name (DN) of the parent object, possible resources:
+  - [aci_tenant](https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/fvTenant) ([fvTenant](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvTenant/overview))
+
+### Read-Only ###
+
+* `id` - (string) The distinguished name (DN) of the Key Ring object.
+* `admin_state` (adminState) - (string) The current administrative state of the certificate request process.
+* `annotation` (annotation) - (string) The annotation of the Key Ring object.
+* `cert` (cert) - (string) A certificate is a file containing a device's public key along with signed information verifying the identity of the device.
+* `description` (descr) - (string) The description of the Key Ring object.
+* `ecc_curve` (eccCurve) - (string) ECC Curve.
```

Comment:
should we rename this attribute elliptic_curve?
should we change the description to be a bit more elaborate?